nemo/collections/asr/data/text_to_text.py

```diff
@@ -0,0 +1,462 @@
+        if isinstance(speakers_filepath, str):
+            speakers_filepath = speakers_filepath.split(",")
+        elif isinstance(speakers_filepath, Path):
+            speakers_filepath = [speakers_filepath]
+        speakers: Set[int] = set()
+        for filepath in speakers_filepath:
+            with open(Path(filepath).expanduser(), "r") as f:
+                speakers.update(map(int, f.read().split()))
+        self.speakers = np.asarray(sorted(speakers))
+        logging.info(f"Loaded {len(self.speakers)} speakers")
+
+        # Load manifest
+        if isinstance(manifest_filepath, str):
+            manifest_filepath = manifest_filepath.split(",")
+        elif isinstance(manifest_filepath, Path):
+            manifest_filepath = [manifest_filepath]
+        self.manifest_paths = [Path(filepath) for filepath in manifest_filepath]
+
+        num_skipped_words = 0
+        num_skipped_utterances = 0
+        asr_texts = []
+        tts_texts = []
+        need_normalization = False
+
+        for manifest_path in self.manifest_paths:
+            for tmp_item in tqdm(_iterate_manifest(manifest_path)):
+                text = tmp_item["text"]
+                num_words = len(text.split())
+                # skip if number of works not in desired range
+                # TODO: maybe it would be valuable to sample sub-utterances from long utterances
+                if not (min_words <= num_words <= max_words):
+                    num_skipped_words += num_words
+                    num_skipped_utterances += 1
+                    continue
+                asr_texts.append(tmp_item["text"])
+                try:
+                    tts_texts.append(tmp_item["tts_text_normalized"])
+                except KeyError:
+                    tts_texts.append(tmp_item["tts_text"])
+                    need_normalization = True
+
+        if need_normalization:
+            logging.warning("TTS normalization is extremely slow! It is recommended to normalize TTS text")
+
+        if num_skipped_utterances:
+            logging.warning(f"Skipped {num_skipped_utterances} utterances " f"with {num_skipped_words}")
+
+        num_utterances = len(asr_texts)
+        # preprocessing is very costly, if we need only part - remove unnecessary utterances
+        if num_parts > 1:
+            # NB: floor division, full dataset can contain fewer utterances than original, like in tarred dataset
+            num_utterances_part = num_utterances // num_parts
+            start = num_utterances_part * current_part_index
+            end = start + num_utterances_part
+            logging.info(
+                f"Taking part of the dataset: {current_part_index} index, total {num_parts} from {start} to {end}"
+            )
+            asr_texts = asr_texts[start:end]
+            tts_texts = tts_texts[start:end]
+            num_utterances = num_utterances_part
+
+        self.data: List[Dict[str, Any]] = [dict() for _ in range(num_utterances)]
```

Comment:
Why are we only initializing it now and not before?

Reply:
Before it's not needed) 
In some cases texts are huge (10s of gigabytes), and loading them in different processes (1 process per gpu!) requires a lot of memory. So, I don't preserve original texts, only necessary tokenized data